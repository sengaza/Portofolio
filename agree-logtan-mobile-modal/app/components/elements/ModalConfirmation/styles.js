import { StyleSheet } from 'react-native';
import { COLOR_WHITE, FONT_BODY3, COLOR_GREY_90, FONT_SUBTITLE1 } from '../../../styles';

const styles = StyleSheet.create({
  button: {
    elevation: 0,
    marginHorizontal: 10,
    width: 100
  },
  container: {
    alignItems: 'center',
    justifyContent: 'space-between',
    backgroundColor: COLOR_WHITE,
    borderRadius: 2,
    height: 180,
    paddingHorizontal: 20,
    paddingVertical: 20,
    width: 270,
    elevation: 5
  },
  descText: {
    ...FONT_BODY3,
    color: COLOR_GREY_90,
    paddingLeft: 20,
    alignSelf: 'flex-start'
  },
  imageWarning: {
    // height: 25,
    // marginTop: 35,
    // width: 25
  },
  titleText: {
    ...FONT_SUBTITLE1,
    color: COLOR_GREY_90,
    paddingLeft: 20,
    alignSelf: 'flex-start'
  }
});

export default styles;
