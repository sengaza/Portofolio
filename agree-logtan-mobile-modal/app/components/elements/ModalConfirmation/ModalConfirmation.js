import React from 'react';
import { View, Text, Image } from 'react-native';
import Modal from 'react-native-modal';
import PropTypes from 'prop-types';
import PrimaryButton from '../PrimaryButton';

import styles from './styles';
import { IMAGES } from '../../../configs';
import { noop } from '../../../utils';
import { COLOR_GREY_40 } from '../../../styles';

export default class ModalConfirmation extends React.Component {
  render() {
    const { title, description, onPressNo, onPressYes, type, leftOption, rightOption } = this.props;
    return (
      <Modal {...this.props} style={{ margin: 0 }} backdropColor={COLOR_GREY_40} backdropOpacity={0.5}>
        <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
          <View style={styles.container}>
            <Image
              source={type === true ? IMAGES.correct : IMAGES.warningYellow}
              style={styles.imageWarning}
              resizeMode="contain"
            />
            <Text style={styles.titleText}>{title}</Text>
            <Text style={styles.descText}>{description}</Text>
            <View style={{ flexDirection: 'row', marginTop: 5 }}>
              {type === true ? null : (
                <PrimaryButton
                  onPress={onPressNo}
                  customContainer={styles.button}
                  size="medium"
                  title={leftOption}
                  type="secondary"
                />
              )}
              <PrimaryButton
                onPress={onPressYes}
                customContainer={styles.button}
                size="medium"
                title={rightOption}
              />
            </View>
          </View>
        </View>
      </Modal>
    );
  }
}

ModalConfirmation.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  onPressNo: PropTypes.func,
  onPressYes: PropTypes.func,
  type: PropTypes.bool,
  leftOption: PropTypes.string,
  rightOption: PropTypes.string
};

ModalConfirmation.defaultProps = {
  title: 'Title',
  description: 'Description',
  onPressNo: noop,
  onPressYes: noop,
  type: false,
  leftOption: 'Tidak',
  rightOption: 'Ya'
};
