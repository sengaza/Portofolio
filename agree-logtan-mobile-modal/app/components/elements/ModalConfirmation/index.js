import ModalConfirmation from './ModalConfirmation';

export default ModalConfirmation;
