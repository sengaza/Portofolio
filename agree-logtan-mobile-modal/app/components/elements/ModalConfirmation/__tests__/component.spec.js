// eslint-disable-next-line no-unused-vars
import { assertSnapshots, mockSetState } from '../../../../utils/TestUtils/snapshot';
import ModalConfirmation from '../ModalConfirmation';

describe('Component snapshot testing', () => {
  const configs = [
    {
      props: {},
      desc: 'renders without data'
    }
  ];
  assertSnapshots(ModalConfirmation, configs);
});

// describe('Component', () => {
//   const ComponentWithMockedState = mockSetState(Component);
//   const getInstance = props => new ComponentWithMockedState(props);
//   beforeEach(() => {
//     this.props = {};
//   });

//   afterEach(() => {
//     this.props = null;
//     jest.clearAllMocks();
//   });

//   test('render on android', () => {
//     jest.mock('Platform', () => {
//       const Platform = require.requireActual('Platform');
//       Platform.OS = 'android';
//       return Platform;
//     });
//     const instance = getInstance(this.props);
//     instance.render();
//   });

//   test('render on ios', () => {
//     jest.mock('Platform', () => {
//       const Platform = require.requireActual('Platform');
//       Platform.OS = 'ios';
//       return Platform;
//     });
//     const instance = getInstance(this.props);
//     instance.render();
//   });
// });
