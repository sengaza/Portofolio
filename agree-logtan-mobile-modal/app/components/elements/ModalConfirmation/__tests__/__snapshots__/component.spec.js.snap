// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot testing renders without data 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={false}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#8C8C8C",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 5,
            "height": 300,
            "paddingHorizontal": 20,
            "width": 300,
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "process": [Function],
            }
          }
          style={
            Object {
              "height": 25,
              "marginTop": 35,
              "width": 25,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#1ea54f",
              "fontFamily": "Quicksand-Medium",
              "fontSize": 31.428571428571427,
              "marginTop": 25,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#8c8c8c",
              "fontFamily": "Quicksand-Medium",
              "fontSize": 18.857142857142858,
              "marginTop": 13,
              "textAlign": "center",
            }
          }
        />
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginTop": 50,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "#1ea54f",
                "borderRadius": 37.714285714285715,
                "borderWidth": 1,
                "elevation": 0,
                "flexDirection": "row",
                "height": 78.47058823529412,
                "justifyContent": "center",
                "marginHorizontal": 10,
                "opacity": 1,
                "width": 100,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#1ea54f",
                    "fontFamily": "Quicksand-Medium",
                    "fontSize": 22,
                  },
                  Object {},
                ]
              }
            >
              Tidak
            </Text>
          </View>
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#1ea54f",
                "borderRadius": 37.714285714285715,
                "elevation": 0,
                "flexDirection": "row",
                "height": 78.47058823529412,
                "justifyContent": "center",
                "marginHorizontal": 10,
                "opacity": 1,
                "width": 100,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontFamily": "Quicksand-Medium",
                    "fontSize": 22,
                  },
                  Object {},
                ]
              }
            >
              Ya
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
